stages/org.osbuild.first-boot

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python3
+
+import json
+import os
+import sys
+
+STAGE_DESC = "Execute commands on first-boot"
+STAGE_INFO = """
+Sequentially execute a list of commands on first-boot / instantiation.
+
+This stage uses a logic similar to systemd's first-boot to execute a given
+script only the first time the image is booted.
+
+An empty flag file /etc/osbuild-first-boot is written to /etc and a systemd
+service is enabled that is only run when the file exits, and will remove it
+before executing the given commands.
+
+If the flag-file cannot be removed, the service fails without executing
+any further first-boot commands.
+"""
+STAGE_OPTS = """
+"required": ["commands"],
+"properties": {
+  "commands": {
+    "type": "array",
+    "description": "The command lines to execute",
+    "items": {
+      "type": "string"
+    }
+  },
+  "wait_for_network": {
+    "type": "bool",
+    "description": "Wait for the network to be up before executing"
+  }
+}
+"""
+
+def add_first_boot(tree, commands, wait_for_network):
+    if wait_for_network:
+        network = """Wants=network-online.target
+After=network-online.target"""
+    else:
+        network = ""
+
+    execs = "\n"
+    for command in commands:
+        execs += f"ExecStart={command}\n"
```

Comment:
We probably want to disallow `\n` in commands or split `command` on newlines.

Reply:
I left this alone for now, it is really not all that straight-forward to determine what is and isn't a valid Exec line.

Reply:
We could improve on this in a follow-up.